pkg/auth/service.go

```diff
@@ -93,6 +95,11 @@ type ExternalPrincipalsService interface {
 	ListUserExternalPrincipals(ctx context.Context, userID string, params *model.PaginationParams) ([]*model.ExternalPrincipal, *model.Paginator, error)
 }
 
+type ServiceAndInviter interface {
```

Comment:
I find this naming too specific...
When reading this, it's hard to figure out when `Service` should be used, and when `ServiceWithInviter` - 
Or in other words, why the email invite capabilities isn't part of the `Service` interface.

Plus, I'd expect to see this interface after the `Service` interface, since it's extending it, but it's probably a matter of personal style.

Reply:
`Service` doesn't require to be an `emailInviter` and currently there are two Implementations of Service:
- `AuthService` - which is not an `emailInviter`
- `APIAuthService` - which is an `emailInviter`
Currently, whether or not the service implements the interface affects the UI, inserting InviteUser kind of breaks the Interface. It is a valid option but then we will need another Method such as IsEmailSender, I considered something like that at first but decided to leave the current implementation. 
@itaigilo do you have an idea for some other implementation or are you suggesting to go with the `isEmailSender` implementation?

Reply:
Maybe I'm missing something, but:
In this case, if we actually expect `APIAuthService` as the `NewMonitoredAuthService()` param - 
Why not changing it to this?
>NewMonitoredAPIAuthService(service APIAuthService)

Reply:
What I'm trying to say, is that we have the general `Service`, which is implemented by `AuthService` and `APIAuthService`,
And now a new sort-of-ad-hoc interface is added, `ServiceAndInviter`, which is an extension of `Service`, but a very minimal one, and which only `APIAuthService` implements. I think it'll be very confusing in the future (meaning, to know when to use each interface / implementation).

Reply:
I'm not sure I follow, 

> to know when to use each interface / implementation

If you want a service Implement the Service, if you want it to send Invites as well implement the EmailSender.
The only reason the `ServiceWithInviter` exists is in order to provide it to the wrapgen tool so it will generate a wrapper that implements both interfaces